### ISSUE TITLE
ObjectStorageClient.copyObject adds "destinationObjectMetadata: null" to the request body

### ISSUE THREAD

**Issue by skichiku** [oracle/oci-java-sdk#282]
I found `ObjectStorageClient.copyObject` added `"destinationObjectMetadata: null"` to the request body as below, although I did NOT set destinationObjectMetadata explicitly.

```
{"sourceObjectName":"test-source","destinationRegion":"xx-xxxxx-x","destinationNamespace":"xxxxxxxxxxxx","destinationBucket":"test-bucket","destinationObjectName":"test-destination","destinationObjectMetadata":null}
```

Please see the minimal github repo [here](https://github.com/skichiku/oci-java-sdk-copyobject-test).

---

I guess `ObjectMetadataInterceptor.intercept(CopyObjectRequest request)` method assigns a empty Map to destinationObjectMetadata, even if I do NOT set destinationObjectMetadata explicitly.
https://github.com/oracle/oci-java-sdk/blob/master/bmc-objectstorage/bmc-objectstorage-generated/src/main/java/com/oracle/bmc/objectstorage/internal/http/ObjectMetadataInterceptor.java#L39 

Probably, this is recognized as '_explicitly set_' in Java SDK and `"destinationObjectMetadata: null"` is added to the request body. 

I was able to bypass the case destinationObjectMetadata is NOT explicitly set by modifying this method like [this](https://github.com/skichiku/oci-java-sdk/commit/8e143d5782f3c4a88299917077c673845a59bf78).

**Comment by SCBob** [comment]
I think the default behavior should be set the destination meta data to be the same as the source metadata (a deep copy), with the option of the caller to override the destination metadata.  At a minimum exposing the REST HeadObject via the JDK would allow a Java Program to get the source metadata to appropriately  set the destination.   Today the only recourse is to get the object, incurring the download of the contents which could be a very large overhead.

**Comment by VidhiBhansali** [comment]
This fix has been merged and released in the latest version of JavaSDK (v1.33.2)
